자바 스크립트 Step2 : 다각형 넓이 구현하기

dev-dongwon opened the pull request:
TUNA, YM 과 함께 페어프로그래밍

dev-dongwon committed:
자바스크립트 2-1 step : 다각형 넓이 구현하기

TUNA, YM 이 함께 페어프로그래밍

dev-dongwon committed:
자바스크립트 step 2-1 : 다각형 넓이 계산기 구현

TUNA, YM 이 함께 페어프로그래밍

dev-dongwon committed:
스텝2 페어 프로그래밍

YM, TUNA 페어프로그래밍

crongro commented on calculator.js:
```diff
@@ -0,0 +1,53 @@
+    }
+}
+
+let checkParamNum = function(...args) {
```
함수들은 const가 좋을까요? let이 좋을까요?

crongro commented on calculators.js:
```diff
@@ -0,0 +1,53 @@
+let getCircleArea = function(r) {
```
커밋하실때 같은 내용의 파일을 두  버전으로 올리지 말고, git commit을 잘 활용하세요.

crongro commented on calculators.js:
```diff
@@ -0,0 +1,53 @@
+}
+
+let getSquareArea = function(width, height) {
+    checkIsNumError(arguments);
```
check 함수 두개를 하나로 묶은 새로운 함수 하나를 추가로 만들어볼래요? 
그래서 여기서는 그 함수를 호출하고요.

참고 : 퍼사드패턴.

dev-dongwon committed:
step2-1 : REVIEW 사항 반영

TUNA & YM 페어프로그래밍 리뷰 사항 반영
- 에러 체크 함수 2개를 단일 기능 함수로 묶음
- 재할당이 필요없는 함수들 모두  let이 아닌 const로 구현

crongro commented on calculators.js:
```diff
@@ -1,53 +1,41 @@
-//console.log(getCylinderArea(undefined));
-//console.log(getCylinderArea(2));
+// 에러 체커는 하나의 함수로 추가
+const errorChecker = function(params, ...args) {
```
👍
